system_tests/manager/test_hostpool_plugin.py

```diff
@@ -13,193 +13,168 @@
-        deployment_id = '{0}-hps-deployment'.format(self.test_id)
-
-        self.upload_deploy_and_execute_install(
-            blueprint_id=blueprint_id,
-            deployment_id=deployment_id,
-            inputs={
-                'centos_image_id': self.env.centos_7_image_id,
-                'windows_image_id': self.env.windows_image_id,
-                'ubuntu_image_id': self.env.ubuntu_trusty_image_id,
-                'flavor_id': self.env.medium_flavor_id,
-                'key_path': '/tmp/{0}.pem'.format(str(uuid.uuid4()))
-            },
-            execute_timeout=EXTENDED_TIMEOUT
-        )
-
-        self.host_pool_service_deployment_id = deployment_id
-
-    def _get_hosts(self, deployment_id):
-        outputs = self.get_outputs(self.host_pool_service_deployment_id)
-        endpoint_url = self.get_endpoint_url(outputs)
-        response = requests.get('{0}/hosts'.format(endpoint_url))
-        self.logger.info('Current hosts: {0}'.format(response.json()))
-        return response.json()
-
-    def _assert_host_state(self, host, state=False):
-        self.assertEquals(host.get('allocated'), state)
-
-    def _scale(self, node_id, delta=+1):
-        self.cfy.execute_workflow('scale', self.host_pool_service_deployment_id,
-                                  parameters=dict(scalable_entity_name=node_id,
-                                                  delta=delta))
-
-    def _uninstall_host_pool_service(self):
-        self.execute_uninstall(self.host_pool_service_deployment_id)
-
-    @property
-    def expected_nodes_count(self):
-        return 5
-
-    def get_inputs(self):
-
-        # the host pool endpoint can be retrieved by getting the deployment
-        # outputs of the host-pool-service deployment
-
-        outputs = self.get_outputs(self.host_pool_service_deployment_id)
-        endpoint_url = self.get_endpoint_url(outputs)
-        inputs = {
-            'host_pool_service_endpoint': endpoint_url
-        }
-        return inputs
-
-    def get_outputs(self, deployment_id):
-        return self.client.deployments.outputs.get(
-            deployment_id=deployment_id).outputs
-
-    def get_endpoint_ip(self, outputs):
-        return outputs['endpoint']['ip_address']
-
-    def get_endpoint_url(self, outputs):
-        endpoint_url = 'http://{0}:{1}'.format(self.get_endpoint_ip(outputs),
-                                               outputs['endpoint']['port'])
-        return endpoint_url
+                self._hostpool.assert_host_state(host, allocated=False)
+
+    def verify_all(self):
+        super(HostpoolNodeCellarExample, self).verify_all()
+        # after uninstalling - which is done at the end of parent class'
+        # verify_all - all the hosts are free again
+        for host in self._hostpool.get_hosts():
+            self._hostpool.assert_host_state(host, allocated=False)
+
+    def assert_nodecellar_working(self, endpoint):
```

Comment:
the original test didn't even check that nodecellar was indeed working on the host, but now we do :)